### PR TITLE
fix: adding private_ipv4_address, ipv6_address to subnet_mapping block

### PR DESCRIPTION
**Pull request by berndherding**
## Description
private_ipv4_address and ipv6_address are passed to the subnet_mapping block of the module

## Motivation and Context
both private_ipv4_address and ipv6_address are parts of the underlying terraform aws_lb resource. like allocation_id,  private_ipv4_address and ipv6_address are optional elements of the subnet_mapping block with no further condition of mutual exclusion. in so far, the passing of these values (with a null default) seems straightforward.

## Breaking Changes
no breaking changes

## How Has This Been Tested?
has been tested with a network load_balancer_type and fixed IPs passed as

```
  subnet_mapping = [
    for subnet_id in var.subnets : {
      "subnet_id" : subnet_id,
      "private_ipv4_address" : module.free_subnet_ip[subnet_id].first_free_ip
    }
  ]
```
(where _free_subnet_ip_ is a module of its own returning the first free ip in a subnet.)
the network loadbalancer was successfully bound to the expected IPs as a result of the test.

### CODE REVIEW

**Comment by Infra-Red** [comment]
@antonbabenko Could you please review this PR? It would be great to be able to configure the IPv6 address for the LBs.

@berndherding Thanks for submitting this pull request! Could you please resolve the conflicts.

File: main.tf [hunk 1]
```diff
@@ -31,8 +31,10 @@ resource "aws_lb" "this" {
     for_each = var.subnet_mapping
 
     content {
-      subnet_id     = subnet_mapping.value.subnet_id
-      allocation_id = lookup(subnet_mapping.value, "allocation_id", null)
+      subnet_id            = subnet_mapping.value.subnet_id
+      allocation_id        = lookup(subnet_mapping.value, "allocation_id", null)
+      private_ipv4_address = lookup(subnet_mapping.value, "private_ipv4_address", null)
```
Comment (antonbabenko) [inline main.tf:36]: This argument was added in `3.1.0` of Terraform AWS provider.

File: main.tf [hunk 2]
```diff
@@ -31,8 +31,10 @@ resource "aws_lb" "this" {
...
+      ipv6_address         = lookup(subnet_mapping.value, "ipv6_address", null)
```
Comment (antonbabenko) [inline main.tf:37]: `ipv6_address` was added in very recent 3.27.0. Please update `versions.tf` to reflect this as a minimum version (in root and in examples).

File: .github/workflows/pre-commit.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+name: Pre-Commit
```
Comment (antonbabenko) [inline .github/workflows/pre-commit.yml:1]: Please replace the content of this file with this one - https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/master/.github/workflows/pre-commit.yml

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Please do these minor fixes and fix conflicts.

**Comment by antonbabenko** [comment]
@berndherding Thank you for this update!

[v5.16.0](https://github.com/terraform-aws-modules/terraform-aws-alb/releases/v5.16.0) has been just released.

**Pull request merged**